src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCreativeEnergy.java

```diff
@@ -104,53 +120,114 @@ public <T> T getCapability(Capability<T> capability, EnumFacing side) {
-                setTier = GTUtility.getTierByVoltage(voltage);
-            }
-        }).setAllowedChars(TextFieldWidget2.NATURAL_NUMS).setMaxLength(19).setValidator(getTextFieldValidator()));
-
-        builder.label(7, 74, "gregtech.creative.energy.amperage");
-        builder.widget(new ClickButtonWidget(7, 87, 20, 20, "-", data -> amps = --amps == -1 ? 0 : amps));
-        builder.widget(new ImageWidget(29, 87, 118, 20, GuiTextures.DISPLAY));
-        builder.widget(new TextFieldWidget2(31, 93, 114, 16, () -> String.valueOf(amps), value -> {
-            if (!value.isEmpty()) {
-                amps = Integer.parseInt(value);
-            }
-        }).setMaxLength(10).setNumbersOnly(0, Integer.MAX_VALUE));
-        builder.widget(new ClickButtonWidget(149, 87, 20, 20, "+", data -> {
-            if (amps < Integer.MAX_VALUE) {
-                amps++;
-            }
-        }));
-
-        builder.dynamicLabel(7, 110, () -> "Energy I/O per sec: " + this.lastEnergyIOPerSec, 0x232323);
-
-        builder.widget(new CycleButtonWidget(7, 139, 77, 20, () -> active, this::setActive,
-                "gregtech.creative.activity.off", "gregtech.creative.activity.on"));
-        builder.widget(new CycleButtonWidget(85, 139, 77, 20, () -> source, value -> {
-            source = value;
+    public @NotNull ModularPanel buildUI(MetaTileEntityGuiData guiData, PanelSyncManager panelSyncManager,
+                                         UISettings settings) {
+        ModularPanel panel = GTGuis.createPanel(this, 176, 140);
+
+        DoubleSyncValue tierSync = SyncHandlers.doubleNumber(() -> setTier, val -> {
+            setTier = (int) val;
+            voltage = GTValues.V[setTier];
+        });
+        LongSyncValue voltageSync = SyncHandlers.longNumber(() -> voltage, val -> {
+            voltage = val;
+            setTier = GTUtility.getTierByVoltage(voltage);
+        });
+        LongSyncValue ampSync = SyncHandlers.longNumber(() -> amps, val -> amps = val);
+        BooleanSyncValue activeSync = SyncHandlers.bool(() -> active, this::setActive);
+        BooleanSyncValue sourceSync = SyncHandlers.bool(() -> source, val -> {
+            source = val;
             if (source) {
-                voltage = 0;
-                amps = 0;
-                setTier = 0;
+                voltage = GTValues.V[GTValues.ULV];
+                amps = 1;
+                setTier = GTValues.ULV;
             } else {
-                voltage = V[MAX];
+                voltage = GTValues.V[MAX];
                 amps = Integer.MAX_VALUE;
                 setTier = MAX;
             }
-        }, "gregtech.creative.energy.sink", "gregtech.creative.energy.source"));
-
-        return builder.build(getHolder(), entityPlayer);
+        });
+
+        panel.child(Flow.column()
+                .margin(7)
+                .crossAxisAlignment(Alignment.CrossAxis.START)
+                .childPadding(4)
+                .child(new SliderWidget()
```

Comment:
This really needs the clicking when sliding along the different voltage values in my opinion.